zq drop query throws error after 4 nested levels.

`$ zq -i zng 'over arr | where id==2 | drop level0.level1.level2.level3' min.zng

panic: duplicate field: "level0.level1.level2.level3"
goroutine 1 [running]:
runtime/debug.Stack()
	/opt/homebrew/Cellar/go@1.17/1.17.8/libexec/src/runtime/debug/stack.go:24 +0x65
github.com/brimdata/zed/runtime/op.(*Catcher).Pull.func1()
	/tmp/zed/runtime/op/catcher.go:25 +0x45
panic({0xa5fee0, 0xc0003343b0})
	/opt/homebrew/Cellar/go@1.17/1.17.8/libexec/src/runtime/panic.go:1038 +0x215
github.com/brimdata/zed/runtime/expr.(*Dropper).newDropper(0xc0004cc880, 0xc0004a62a0, 0xc000010000)
	/tmp/zed/runtime/expr/dropper.go:66 +0x278
github.com/brimdata/zed/runtime/expr.(*Dropper).Eval(0xc0004cc880, {0x7fb3acc83488, 0xc0004a70b0}, 0xc000010000)
	/tmp/zed/runtime/expr/dropper.go:119 +0x9e
github.com/brimdata/zed/runtime/op.(*applier).Pull(0xc0004decf0, 0x8)
	/tmp/zed/runtime/op/apply.go:34 +0x1a2
github.com/brimdata/zed/runtime/op.(*Single).Pull(0xc00049fd40, 0x20)
	/tmp/zed/runtime/op/mux.go:138 +0x35
github.com/brimdata/zed/runtime/op.(*Catcher).Pull(0xa7d018, 0xe0)
	/tmp/zed/runtime/op/catcher.go:28 +0x6d
github.com/brimdata/zed/runtime.(*Query).Pull(0x9fa3de, 0x0)
	/tmp/zed/runtime/query.go:106 +0x45
github.com/brimdata/zed/zbuf.(*pullerReader).Read(0xc00047af00)
	/tmp/zed/zbuf/batch.go:131 +0x42
github.com/brimdata/zed/zbuf.(*noControl).Read(0xc0004d8aa0)
	/tmp/zed/zbuf/reader.go:31 +0x30
github.com/brimdata/zed/zio.CopyWithContext({0xcb8790, 0xc000194000}, {0x7fb3acc3f678, 0xc00049f248}, {0xcab460, 0xc0004d8aa0})
	/tmp/zed/zio/zio.go:150 +0x62
github.com/brimdata/zed/zio.Copy(...)
	/tmp/zed/zio/zio.go:142
github.com/brimdata/zed/cli/zq.(*Command).Run(0xc0002ce300, {0xc000192030, 0x2, 0x2})
	/tmp/zed/cli/zq/command.go:168 +0x81c
github.com/brimdata/zed/pkg/charm.path.run({0xc0001800b8, 0x1, 0x1}, {0xc000192030, 0x0, 0x0})
	/tmp/zed/pkg/charm/path.go:11 +0x8e
github.com/brimdata/zed/pkg/charm.(*Spec).ExecRoot(0xc000056750, {0xc000192010, 0x4, 0x4})
	/tmp/zed/pkg/charm/charm.go:63 +0x4a
main.main()
	/tmp/zed/cmd/zq/main.go:11 +0x5f`

However anything less than 4 levels works as expected.
Cases that work:
$ zq -i zng 'over arr | where id==2 | drop level0.level1.level2' min.zng
$ zq -i zng 'over arr | where id==2 | drop level0.level1' min.zng

Cases that **doesn't** work: 
$ zq -i zng 'over arr | where id==2 | drop level0.level1.level2.level3' min.zng
$ zq -i zng 'over arr | where id==2 | drop level0.level1.level2.level3.level4' min.zng
$ zq -i zng 'over arr | where id==2 | drop level0.level1.level2.level3.level4.level5' min.zng

Anything more than level3 (4th level) displays same error.

panic: duplicate field: "level0.level1.level2.level3"

Attaching min.zng file.
[min.zng.zip](https://github.com/brimdata/zed/files/8669599/min.zng.zip)

--- a/runtime/expr/dropper.go
+++ b/runtime/expr/dropper.go
@@ -91,7 +91,7 @@ func complementFields(drops field.List, prefix field.Path, typ *zed.TypeRecord) 
 				continue
 			}
 		}
-		fields = append(fields, append([]string{}, fld...))
+		fields = append(fields, append(field.Path{}, fld...))
 		types = append(types, c.Type)
 	}
 	return fields, types, match

--- a/runtime/expr/dropper.go
+++ b/runtime/expr/dropper.go
@@ -78,31 +78,23 @@ func complementFields(drops field.List, prefix field.Path, typ *zed.TypeRecord) 
 	var types []zed.Type
 	var match bool
 	for _, c := range typ.Columns {
-		if contains(drops, append(prefix, c.Name)) {
+		fld := append(prefix, c.Name)
+		if drops.Has(fld) {
 			match = true
 			continue
 		}
 		if typ, ok := zed.TypeUnder(c.Type).(*zed.TypeRecord); ok {
-			if fs, ts, m := complementFields(drops, append(prefix, c.Name), typ); m {
+			if fs, ts, m := complementFields(drops, fld, typ); m {
 				fields = append(fields, fs...)
 				types = append(types, ts...)
 				match = true
 				continue
 			}
 		}
-		fields = append(fields, append(prefix, c.Name))
+		fields = append(fields, append([]string{}, fld...))
 		types = append(types, c.Type)
 	}
 	return fields, types, match
-}
-
-func contains(ss field.List, el field.Path) bool {
-	for _, s := range ss {
-		if s.Equal(el) {
-			return true
-		}
-	}
-	return false
 }
 
 func (_ *Dropper) String() string { return "drop" }

--- a/runtime/expr/ztests/drop-nested.yaml
+++ b/runtime/expr/ztests/drop-nested.yaml
@@ -2,6 +2,8 @@ script: |
   zq -z "drop rec.bar" nested1.zson
   echo ===
   zq -z "drop rec1" nested2.zson
+  echo ===
+  zq -z "drop level0.level1.level2.level3" nested3.zson
 
 inputs:
   - name: nested1.zson
@@ -13,6 +15,19 @@ inputs:
     data: |
       {foo:"outer1",rec1:{sub1:{foo:"foo1.1",bar:"bar1.1"},sub2:{foo:"foo2.1",bar:"bar2.1"}},rec2:{foo:"foo3.1"}}
       {foo:"outer2",rec1:{sub1:{foo:"foo1.2",bar:"bar1.2"},sub2:{foo:"foo2.2",bar:"bar2.2"}}(=rec1_named),rec2:{foo:"foo3.2"}(=rec2_named)}(=named)
+  - name: nested3.zson
+    data: |
+      {
+        level0: {
+          level1: {
+            level2: {
+              level3: "hi",
+              foo: 1,
+              bar: 2
+            }
+          }
+        }
+      }
 
 outputs:
   - name: stdout
@@ -23,3 +38,5 @@ outputs:
       ===
       {foo:"outer1",rec2:{foo:"foo3.1"}}
       {foo:"outer2",rec2:{foo:"foo3.2"}(=rec2_named)}
+      ===
+      {level0:{level1:{level2:{foo:1,bar:2}}}}

